conversions/src/types.rs

```diff
@@ -0,0 +1,239 @@
+        let private_key = PrivateKey::from_plonky2_secret_key(plonky2_sk).unwrap();
+        // assert that the [`BigUint`] representations match
+        assert_eq!(
+            BigUint::from_bytes_le(
+                &plonky2_sk
+                    .0
+                     .0
+                    .iter()
+                    .flat_map(|u| u.to_le_bytes())
+                    .collect::<Vec<_>>()
+            ),
+            BigUint::from_bytes_be(&private_key.0.serialize())
+        );
+    }
+
+    #[test]
+    fn it_works_plonky2_libsecp256k1_public_key_coordinates() {
+        let plonky2_sk = ECDSASecretKey::<Secp256K1>(Secp256K1Scalar::rand());
+        let private_key = PrivateKey::from_plonky2_secret_key(plonky2_sk).unwrap();
+        let plonky2_public_key = plonky2_sk.to_public();
+        let public_key = libsecp256k1::PublicKey::from_secret_key(&private_key.0);
+
+        // assert that both x coordinates of public keys match
+        assert_eq!(
+            BigUint::from_bytes_be(&public_key.serialize()[1..33]),
+            plonky2_public_key.0.x.to_canonical_biguint()
+        );
+        // assert that both y coordinates of public key match
+        assert_eq!(
+            BigUint::from_bytes_be(&public_key.serialize()[33..65]),
+            plonky2_public_key.0.y.to_canonical_biguint()
+        );
+    }
+
+    #[test]
+    fn it_works_pubkey_from_plonky2_sk() {
+        let plonky2_sk = ECDSASecretKey::<Secp256K1>(Secp256K1Scalar::rand());
+        let public_key = PublicKey::from_plonky2_secret_key(&plonky2_sk).unwrap();
+        let should_be_public_key =
+            PublicKey::from_private_key(&PrivateKey::from_plonky2_secret_key(plonky2_sk).unwrap());
+        assert_eq!(public_key, should_be_public_key);
+    }
+
+    #[test]
+    fn it_works_plonky2_libsecp256k1_public_key_conversion() {
+        let plonky2_sk = ECDSASecretKey::<Secp256K1>(Secp256K1Scalar::rand());
+        let private_key = PrivateKey::from_plonky2_secret_key(plonky2_sk).unwrap();
+        let plonky2_public_key = plonky2_sk.to_public();
+        let public_key = PublicKey::from_plonky2_public_key(plonky2_public_key).unwrap();
+        let should_be_public_key = PublicKey::from_private_key(&private_key);
+        // assert that both x coordinates of public keys match
+        assert_eq!(public_key, should_be_public_key);
+    }
+
+    #[test]
+    fn it_works_libsecp256k1_plonky2_public_key_conversion() {
+        let plonky2_sk = ECDSASecretKey::<Secp256K1>(Secp256K1Scalar::rand());
+        let private_key = PrivateKey::from_plonky2_secret_key(plonky2_sk).unwrap();
+        let should_be_plonky2_public_key = plonky2_sk.to_public();
+        let public_key = PublicKey::from_private_key(&private_key);
+        let plonky2_public_key = public_key.into_plonky2_public_key();
+        assert_eq!(plonky2_public_key, should_be_plonky2_public_key);
+    }
+
+    #[test]
+    fn it_works_signature_scheme_libsecp256k1_plonky2_conversion() {
```

Comment:
this is a bit of a tangle between crates; fine for now

Id either copy the helper functions into mod test here, or move this test to the eth-wallet crate?

Reply:
actually it is here too https://github.com/eureka-network/Solina/pull/1/files#diff-9490abcbb46760b2f07c72ddb8cb1b4e38408f32b8a98b98040dcc4ac1b721c6R66


so maybe this can be deleted and backwards dependencies (on wallet) removed?

Reply:
Nice, refactored it accordingly